aptos-move/block-executor/src/view.rs

```diff
@@ -545,10 +534,45 @@ impl<'a, T: Transaction, X: Executable> ParallelState<'a, T, X> {
 
         loop {
             match self.versioned_map.data().fetch_data(key, txn_idx) {
-                Ok(Versioned(version, v, layout)) => {
-                    let data_read = DataRead::Versioned(version, v.clone(), layout)
+                Ok(Versioned(version, value)) => {
+                    // If we have a known layout, upgrade RawFromStorage value to Exchanged.
+                    if let UnknownOrLayout::Known(layout) = layout {
+                        if let ValueWithLayout::RawFromStorage(v) = value {
+                            match patch_base_value(v.as_ref(), layout) {
```

Comment:
let's assert that the version is storage version?

Reply:
Done.